File: 10) Java OOP - February2019/09. Inheritance - Exercises/01. Person/people/Child.java
package people;


public class Child extends Person {

    private static final int MIN_AGE = 15;


    public Child(String name, int age) throws IllegalArgumentException {
        super(name, age);
    }

    @Override
    public void setAge(int age) throws IllegalArgumentException {
        if (age > MIN_AGE) {
            throw new IllegalArgumentException("Child's age must be lesser than 15!");
        }
        super.setAge(age);
    }
}

File: Software University/04) PHP Web Development Basics/12) OOP FUNDAMENTALS - Exercises/04. Person/Child.php
<?php


namespace Van\Classes\Fourth;


class Child extends Person
{
    public function __construct($name, $age)
    {
        parent::setName($name);
        $this->setAge($age);
    }

    protected function setAge(int $age): Person
    {
        if ($age > 15) {
            throw new \Exception('Child\'s age must be less than 16!');
        }

        parent::setAge($age);

        return $this;
    }
}